README.md

```diff
@@ -187,3 +187,101 @@ After that you can start QEMU
+kas-container build meta-trenchboot/kas-generic-tb.yml
+```
+
+Replace `<recipes>` with name of changed recipes e.g. `skl`, `grub`,
+`linux-tb` and then rebuild.
+
+### Source revision
+
+To change branch or commit used by a recipe you have to change `BRANCH` or
+`SRCREV` variable in appropriate recipe file. In case of Linux kernel those
+variables are named `KBRANCH` and `SRCREV_machine`
+
+Below is list of recipe files for main TB components
+
+```text
+intel-sinit-acm - recipes-support/intel-sinit-acm/intel-sinit-acm_630744.bb
+skl - recipes-support/skl/skl_git.bb
+linux-tb - recipes-kernel/linux/linux-tb_6.6.bb
+grub & grub-efi (common parts) - recipes-bsp/grub/grub-tb-common.inc
+grub - recipes-bsp/grub/grub_%.bbappend
+grub-efi - recipes-bsp/grub/grub-efi_%.bbappend
+```
+
+### Building modified source
+
+In order to make and test changes to source code you first need to enter kas
+shell
+
+```shell
+kas-container shell meta-trenchboot/kas-generic-tb.yml
+```
+
+After that use `devtool modify <recipe>` to fetch source code of recipe you want
+to work on. Bitbake will checkout branch and commit set by `BRANCH` and `SRCREV`
+variables.
+
+```shell
+devtool modify skl
+(...)
+INFO: Source tree extracted to /build/workspace/sources/skl
+INFO: Using source tree as build directory since that would be the default for this recipe
+INFO: Recipe skl now set up to build from /build/workspace/sources/skl
+```
+
+All recipes' sources you wish to modify will be in `/build/workspace/sources`.
+After modifications you can try to build recipe by using
+`devtool build <recipe>` or if you want to build whole image with your changes
+then use `devtool build-image tb-minimal-image`.
+After building image you can [flash](#flash) and [boot](#booting) it or run it
+in [QEMU](#running-in-qemu).
+
+### Saving changes
+
+Commit changes you want saved and then use `devtool update-recipe <recipe>`.
+Devtool should generate `.patch` file and save it in correct place and also
+update recipe file.
+
+```shell
+builder@2906c1d05d2c:/build/workspace/sources/skl$ devtool update-recipe skl
+(...)
+INFO: Adding new patch 0001-wip.patch
+INFO: Updating recipe skl_git.bb
+```
+
+#### GRUB
+
+In case of GRUB you should use `devtool update-recipe -a /repo -w <recipe>`
+where `<recipe>` is either `grub` or `grub-efi`. This is due to original
+recipe(`.bb` file) being defined in dfferent layer.
+
+```shell
+builder@2906c1d05d2c:/build/workspace/sources$ devtool update-recipe -a /repo -w grub-efi
+(...)
+Summary: There was 1 WARNING message.
+NOTE: Writing append file /repo/recipes-bsp/grub/grub-efi_%.bbappend
+NOTE: Copying 0001-wip.patch to /repo/recipes-bsp/grub/grub-efi/0001-wip.patch
+```
+
+### Local files
+
+Files added by Yocto recipe are stored inside `sources/<recipe>/oe-local-files`
+folder. When you use `devtool update-recipe` those files are automatically
+updated without committing them. Example of local file is `grub.cfg` in
+`grub-efi` recipe
+
```

Comment:
Added information how to deploy component changes: 13ee5090e5194201df784ff74d4ed7e14d774e56
If it's possible to boot into TrenchBoot then changes could be copied via ssh otherwise copy changes to mounted partition from flash drive.

Reply:
I think that scenario makes sense. Even if you cannot boot from Trenchboot, you can have another backup working system, to which you can always login and deploy directly to TB disk via this another system.
I think that was also the approach here https://github.com/TrenchBoot/trenchboot-sdk/blob/master/scripts/aem/aem.sh#L238